pkgs/by-name/op/opengamepadui/package.nix

```diff
@@ -0,0 +1,117 @@
+{
+  alsa-lib,
+  autoPatchelfHook,
+  cargo,
+  dbus,
+  fetchFromGitHub,
+  gamescope,
+  gitUpdater,
+  godot_4,
+  godot_4-export-templates,
+  hwdata,
+  lib,
+  libGL,
+  libpulseaudio,
+  mesa-demos,
+  pkg-config,
+  rustPlatform,
+  stdenv,
+  udev,
+  upower,
+  vulkan-loader,
+  xorg,
+  withDebug ? false,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "opengamepadui";
+  version = "0.35.4";
+
+  buildType = if withDebug then "debug" else "release";
+
+  src = fetchFromGitHub {
+    owner = "ShadowBlip";
+    repo = "OpenGamepadUI";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-YjXGrE2QVKMLgYy1Tv8QPGgSfqZFsr+G17qfjYQXLFg=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit (finalAttrs) src;
+    sourceRoot = "source/${finalAttrs.cargoRoot}";
+    hash = "sha256-sTzMewIfKHbmVhSPZgUIzFFz1ahK+PMoQ5oB4GEt8nY=";
+  };
+  cargoRoot = "extensions";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    cargo
+    godot_4
+    godot_4-export-templates
+    pkg-config
+    rustPlatform.cargoSetupHook
+  ];
+
+  runtimeDependencies = [
+    alsa-lib
+    dbus
+    gamescope
+    hwdata
+    libGL
+    libpulseaudio
+    mesa-demos
+    udev
+    upower
+    vulkan-loader
+    xorg.libX11
+    xorg.libXcursor
+    xorg.libXext
+    xorg.libXi
+    xorg.libXrandr
+    xorg.libXres
+    xorg.libXtst
+  ];
+
+  dontStrip = withDebug;
+
+  env =
+    let
+      versionAndRelease = lib.splitString "-" godot_4.version;
+    in
+    {
+      GODOT = lib.getExe godot_4;
+      GODOT_VERSION = lib.elemAt versionAndRelease 0;
+      GODOT_RELEASE = lib.elemAt versionAndRelease 1;
+      EXPORT_TEMPLATE = "${godot_4-export-templates}";
+      BUILD_TYPE = "${finalAttrs.buildType}";
+    };
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  buildFlags = [ "build" ];
+
+  preBuild = ''
+    # Godot looks for export templates in HOME
+    export HOME=$(mktemp -d)
+    mkdir -p $HOME/.local/share/godot/export_templates
+    ln -s "${godot_4-export-templates}" "$HOME/.local/share/godot/export_templates/$GODOT_VERSION.$GODOT_RELEASE"
+  '';
+
+  postInstall = ''
+    # The Godot binary looks in "../lib" for gdextensions
+    mkdir -p $out/share/lib
+    mv $out/share/opengamepadui/*.so $out/share/lib
+  '';
+
+  passthru.updateScript = gitUpdater { rev-prefix = "v"; };
+
+  meta = {
+    description = "Open source gamepad-native game launcher and overlay";
+    homepage = "https://github.com/ShadowBlip/OpenGamepadUI";
+    license = lib.licenses.gpl3Only;
+    platforms = lib.platforms.linux;
```

Comment:
The Makefile of OpenGamepadUI is tied to x86_64, so platforms should be restricted to just `x86_64-linux`.

https://github.com/ShadowBlip/OpenGamepadUI/blob/dccb96705cc622948dbe9f09008d808c3d69b384/Makefile#L111

Reply:
It compiles on aarch64-linux, but it generates an x86_64-linux shared object:

```
$ file /nix/store/*-opengamepadui-0.35.4/share/opengamepadui/opengamepad-ui.x86_64
/nix/store/0x13bbsy7jik93kh1ilkx5w2chf6sjx2-opengamepadui-0.35.4/share/opengamepadui/opengamepad-ui.x86_64: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 5.15.0, stripped
```

Reply:
Arm support should be possible, but adding support for it will require a bit of work to factor out all of the architecture assumptions. Ideally OpenGamepadUI should also be paired with Gamescope, which I'm not sure works very well on arm at the moment.

I will update this to be `x86_64-linux` only for now and hopefully can add arm support in the future.

Reply:
I briefly looked into this. It looks like Godot right now only supports compiling to x86-64.

Reply:
I briefly looked into this. It looks like Godot right now only supports compiling to x86-64.